src/auth/auth.controller.ts

```diff
@@ -51,42 +57,45 @@ export class AuthController {
   }
 
   @Get('refresh')
-  async refresh(@Req() request, @Res({ passthrough: true }) response){
-    const tokenFromUser = request.cookies.refreshToken;
+  async refresh(@Req() request, @Res({ passthrough: true }) response, @SystemInfo() systemInfo){
 
-    const { refreshToken, ...rest } = await this.authService.refreshSession(tokenFromUser);
+    const { refreshToken, ...rest } = await this.authService.refreshSession(request.cookies.refreshToken, systemInfo);
 
-    response.cookie( 'refreshToken', refreshToken );
+    response.cookie('refreshToken', refreshToken);
     return rest;
   }
 
+  @Get('sessions')
+  @UseGuards(TokenAuthGuard)
+  getSessions(@Req() request){
+    return this.sessionHandler.getAllSession(request.user.id);
+  }
+
   @Get('oauth/google')
   @UseGuards(AuthGuard('google'))
-  async googleAuth(@Req() req){}
+  googleAuth(){}
```

Comment:
for what this function?

Reply:
this function is required for AuthGuard to be able to redirect user in case of successful authentication